balancer_conn_wrappers.go

```diff
@@ -243,7 +243,9 @@ func (ccb *ccBalancerWrapper) UpdateBalancerState(s connectivity.State, p balanc
 	// case where we wait for ready and then perform an RPC.  If the picker is
 	// updated later, we could call the "connecting" picker when the state is
 	// updated, and then call the "ready" picker after the picker gets updated.
-	ccb.cc.blockingpicker.updatePicker(p)
+	if ccb.cc.blockingpicker.picker != p {
```

Comment:
It's by design to update picker even if it's the same one.

For each RPC, if there's no ready connection to be used (all are connecting), they will block. Picker update is used as a signal for those RPCs to pick again. It's not required for the new picker to be different from the previous one.